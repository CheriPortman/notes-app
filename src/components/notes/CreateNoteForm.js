import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class CreateNoteForm extends PureComponent {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired
  };

  state = { title: '', text: '' };

  handleSubmit = event => {
    event.preventDefault();
    const { title, text } = this.state;

    this.props.onSubmit(title, text);
    this.setState({ title: '', text: '' });
  }

  

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { title, text } = this.state;
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="title">
          Name your note:
          <input name="title" type="text" value={title} onChange={this.handleChange} />
        </label>
        <label htmlFor="text">
          Enter the text of your note:
          <textarea name="text" type="text" value={text} onChange={this.handleChange} />
        </label>
        <button>Submit</button>
      </form>
    );
  }
}
