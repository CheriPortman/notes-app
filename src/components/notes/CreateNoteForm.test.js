import React from 'react';
import { shallow } from 'enzyme';
import CreateNoteForm from './CreateNoteForm';

describe('CreateNoteForm componenet', () => {
  it('renders the form', () => {
    const wrapper = shallow(<CreateNoteForm onSubmit={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('updates title on change', () => {
    const wrapper = shallow(<CreateNoteForm onSubmit={jest.fn()} />);
    wrapper.find('[name="title"]').at(1).simulate('change', {
      target: { name: 'title', value: 'note' }
    });
  });

  it('updates text on change', () => {
    const wrapper = shallow(<CreateNoteForm onSubmit={jest.fn()} />);
    wrapper.find('[name="text"]').at(0).simulate('change', {
      target: { name: 'text', value: 'this is text' }
    });
  });

  it('handles submitting a new note', () => {
    const submit = jest.fn();
    const wrapper = shallow(<CreateNoteForm onSubmit={submit} />);

    wrapper.find('form').at(1).simulate('submit', {
      preventDefault: jest.fn(),
    });

    expect(submit).toHaveBeenCalled();
  });
});
