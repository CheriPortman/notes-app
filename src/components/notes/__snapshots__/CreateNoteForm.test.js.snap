// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNoteForm componenet renders the form 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.label
    htmlFor="title"
  >
    Name your note:
    <styled.input
      name="title"
      onChange={[Function]}
      type="text"
      value=""
    />
  </styled.label>
  <styled.label
    htmlFor="body"
  >
    Enter the text of your note:
    <styled.input
      name="body"
      onChange={[Function]}
      type="text"
      value=""
    />
  </styled.label>
  <styled.button>
    Submit
  </styled.button>
</styled.form>
`;
